src/main/java/org/stellar/sdk/Server.java

```diff
@@ -115,41 +109,18 @@ public TransactionsRequestBuilder transactions() {
      * @throws IOException
      */
     public SubmitTransactionResponse submitTransaction(Transaction transaction) throws IOException {
-        URI transactionsURI;
-        try {
-            transactionsURI = new URIBuilder(serverURI).setPath("/transactions").build();
-        } catch (URISyntaxException e) {
-            throw new AssertionError(e);
-        }
-        HttpPost submitTransactionRequest = new HttpPost(transactionsURI);
-
-        List<NameValuePair> params = new ArrayList<NameValuePair>();
-        params.add(new BasicNameValuePair("tx", transaction.toEnvelopeXdrBase64()));
-        submitTransactionRequest.setEntity(new UrlEncodedFormEntity(params));
-
-        HttpResponse response = httpClient.execute(submitTransactionRequest);
-        HttpEntity entity = response.getEntity();
-
-        if (entity != null) {
-            InputStream responseStream = entity.getContent();
-            try {
-                StringWriter writer = new StringWriter();
-                IOUtils.copy(responseStream, writer, StandardCharsets.UTF_8);
-                String responseString = writer.toString();
-                SubmitTransactionResponse submitTransactionResponse = GsonSingleton.getInstance().fromJson(responseString, SubmitTransactionResponse.class);
-                return submitTransactionResponse;
-            } finally {
-                responseStream.close();
-            }
+        HttpUrl transactionsURI = serverURI.newBuilder().addPathSegment("transactions").build();
+        RequestBody requestBody = new FormBody.Builder().add("tx", transaction.toEnvelopeXdrBase64()).build();
+        Request submitTransactionRequest = new Request.Builder().url(transactionsURI).post(requestBody).build();
+        Response response = this.httpClient.newCall(submitTransactionRequest).execute();
+
+        ResponseBody responseBody = response.body();
+
+        if (responseBody == null) {
```

Comment:
In that case the body is never closed and will lead to a memory leak

Reply:
You can't close `ResponseBody` which is `null`. And if it's not `null`, it's automatically closed because we're using `string()` method. This has been explained in another comment in this PR.